plugins/secrets/index.js

```diff
@@ -58,6 +58,10 @@ exports.register = (server, options, next) => {
                 });
             }
 
+            if (scope.includes('pipeline') && secret.pipelineId !== credentials.pipelineId) {
```

Comment:
I think the code below is more concise.
```
if (secret.pipelineId !== credentials.pipelineId) {
  if (scope.includes('pipeline') {
    throw boom.forbidden('Token does not have permission to this secret');
  }
  if (secret.pipelineId !== credentials.configPipelineId) {
    throw boom.forbidden(`${username} is not allowed to access this secret`);
  }
}
```

Reply:
I think the above code is a little bit confusing to read. In L44, this method checks if token's scope is user or not. Therefore, it is natural to check if token's scope is pipeline or not first.

Reply:
I see, I think that's fine.